### PR TITLE
DB modules hierarchy restructuring and renaming and relative Sqlite.hs code split

### PR DESCRIPTION
**Pull request by paolino**
### Motivation 

I want to push a `no ticket value`  prework on the DB modules structure to put some standard names to things and create a more semantically thought modules layout.
It is an effort that we can ditch or make it worth with even better structure that I cannot foresee. 

This is master 

```
.
├── Checkpoints.hs
├── Model.hs
├── MVar.hs
├── Sqlite
│   ├── AddressBook.hs
│   ├── CheckpointsOld.hs
│   ├── Migration.hs
│   ├── Stores.hs
│   ├── TH.hs
│   └── Types.hs
├── Sqlite.hs
└── WalletState.hs
```

This is the proposed reshuffle

```
├── Checkpoints (new namespace for a component)
│   ├── AddressBook.hs (was Sqlite/AddressBook.hs) 
│   ├── Model.hs (types + pure functionality from CheckpointsOld.hs)
│   └── Store.hs (persistence code from CheckpointsOld.hs)
├── Pure
│   ├── Layer.hs (ex MVar.hs)
│   └── Implementation.hs (ex Model.hs)
├── Layer (ex Sqlite.hs)
├── Sqlite
│   ├── Migration.hs (no changes)
│   ├── Schema.hs (was TH.hs)
│   ├── Types.hs (no changes)
│   └── WrapSTM.hs (was buried somewhere needed)
├── Unstored.hs (part of Sqlite.hs still to be splitted in components)
└── Wallets
    ├── State.hs (was WalletSate.hs)
    └── Store.hs (was Sqlite/Stores.hs)

```

- [x] rename TH to Schema
> every database layer should have a schema definition module. TH has no semantic relevance
- [x] rename Sqlite to Layer
> The code is about the DBLayer instance and code is SQLite agnostic 
- [x] move AddressBook into Checkpoints space
> AddressBook is a necessary component  only of Checkpoints (to be proved really)
- [x] move Checkpoints code into Checkpoints space
> we want to redesign by compartmentalizing in stores. One store is Checkpoints
- [x] move WrapSTM into Sqlite space (it's Persistent relevant) 
> this is a library component, should go as sideway as possible
- [x] move WalletsState.hs into Wallets/State.hs  
> Wallets is a DB component  that depends on the other components and has a composed State
- [x] move Wallets Store relevant code  into Wallets.Store module
> Wallets is a DB component  that depends on the other components and has a composed Store
- [x] move MVar.hs into Pure/Layer.hs 
> this is the implementation of a pure DBLayer
- [x] move Model into Pure/Implementation.hs
> this is an implementation of a pure (in memory) DB functionality for testing purposes

### Comments

It's mostly cosmetic but it gives some more structure to explore the Store design  

### Issue Number

ADP-1784, ADP-1043

### CODE REVIEW

File: lib/core/src/Cardano/Wallet/DB.hs [hunk 1]
```diff
@@ -358,6 +359,8 @@ newtype ErrNoSuchWallet
     = ErrNoSuchWallet WalletId -- Wallet is gone or doesn't exist yet
     deriving (Eq, Show)
 
+instance Exception ErrNoSuchWallet
```
Comment (HeinrichApfelmus) [inline lib/core/src/Cardano/Wallet/DB.hs]: To make review possible, please only rearrange modules during this Pull Request — additions and deletions of stuff go into a separate Pull Request. 🤓

As it turns out, this `instance Exception ErrNoSuchWallet` is surprisingly controversial. 😱 It has ramifications for how we deal with errors/exceptions in our code.

File: lib/core/src/Cardano/Wallet/DB/Checkpoints/Model.hs [hunk 2]
```diff
@@ -8,38 +9,54 @@
...
 -- Data type that represents a collection of checkpoints.
 -- Each checkpoints is associated with a 'Slot'.
+module Cardano.Wallet.DB.Checkpoints.Model
+  ( -- * Checkpoints
+    Checkpoints,
+    checkpoints,
+    loadCheckpoints,
+    fromGenesis,
+    getLatest,
+    findNearestPoint,
```
Comment (HeinrichApfelmus) [inline lib/core/src/Cardano/Wallet/DB/Checkpoints/Model.hs]: The formatting of this module got a little messed up. 🤔 Could you restore it to how it was and restrict the change to the module name?

Current style guides:
* Indentation is 4 spaces by default. (Exception: 2 spaces for a single-line `where` keyword)
* Import lists, lists, and records have the "commas first" style.

@jonathanknowles

File: lib/core/src/Cardano/Wallet/DB/Wallets/Store.hs [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+-- Copyright: © 2022 IOHK
+-- License: Apache-2.0
+--
+-- 'Store' implementations that can store various wallet types
+-- in an SQLite database using `persistent`.
+--
+-- FIXME LATER during ADP-1043:
+--
+-- * Inline the contents of "Cardano.Wallet.DB.Sqlite.CheckpointsOld"
+--   into this module
```
Comment (HeinrichApfelmus) [inline lib/core/src/Cardano/Wallet/DB/Wallets/Store.hs]:
```suggestion
```

Comment is acted upon through this Pull Request.

File: lib/core/src/Cardano/Wallet/Primitive/Types/Tx.hs [hunk 4]
```diff
@@ -244,6 +244,7 @@ data Tx = Tx
 
     , collateralOutput :: !(Maybe TxOut)
         -- ^ An output that is only created if a transaction script fails.
+            -- Q: created means spent here ? Isn't this either the 'outputs' ?
```
Comment (HeinrichApfelmus) [inline lib/core/src/Cardano/Wallet/Primitive/Types/Tx.hs]:
```suggestion
```
A: This output is added to the ledger ("created") if and only if this transaction is accepted into the ledger even though script validation fails due to insufficient fees (in this event, the ordinary outputs are not created in the ledger.) The word "fail" is a bit overloaded here — the transaction does make it into the ledger (no fail), but script validation fails due to insufficient fees.

Spending a transaction output only happens when they are referenced in the inputs of a transaction.

File: lib/dbvar/src/Data/DeltaMap.hs [hunk 5]
```diff
@@ -0,0 +1,28 @@
+{-# LANGUAGE TypeFamilies #-}
+
+module Data.DeltaMap where
```
Comment (HeinrichApfelmus) [inline lib/dbvar/src/Data/DeltaMap.hs]:
```suggestion
module Data.DeltaMap
    ( DeltaMap(..)
    ) where
```

**Review by HeinrichApfelmus — COMMENTED** [review]
Looks almost good to me! 😊 Before approval, could you fix the formatting of `Cardano.Wallet.DB.Checkpoints.Model` and remove the `instance Exception ErrNoSuchWallet`?

File: lib/core/src/Cardano/Wallet/DB.hs (hunk 1 above)
Comment (paolino) [reply lib/core/src/Cardano/Wallet/DB.hs]: how SomeException is supposed to be used for `loadS :: Either SomeException a`? In this case, I think asking for a non-stored wallet id has already an error, namely `ErrNoSuchWallet`. I feel introducing a new duplicate error is not what we want. What is actually controversial here?  I removed it for now as still not needed

Comment (HeinrichApfelmus) [reply lib/core/src/Cardano/Wallet/DB.hs]: > What is actually controversial here?

Long story. In the wallet, we use a lot `ExceptT e IO`. This is an example of a "checked exception" where the exception type `e` is visible on the type level. But in order for this to make sense, `e` should _not_ be an instance of the `Exception` class — to guarantee that `e` cannot be thrown by the `IO` portion. That's the main reason.

When designing `loadS :: Either SomeException a`, I intentionally did not use checked exceptions in order to simplify the types (one less type parameter). But this is somewhat at odds with the checked exceptions style of `ExceptT`.

(My personal preference, and common recommendation, is to not use `ExceptT`. I feel that the time spent on converting different error types into each other could be spent on making the code more correct instead.)

File: lib/core/src/Cardano/Wallet/DB/Checkpoints/Model.hs [hunk 6]
```diff
@@ -100,35 +120,73 @@ fromGenesis a = Checkpoints $ Map.singleton W.Origin a
...
+        Wallet Checkpoint
+-------------------------------------------------------------------------------}
+
+-- | Data stored in a single checkpoint.
+-- Only includes the 'UTxO' and the 'Discoveries', but not the 'Prologue'.
+data WalletCheckpoint s = WalletCheckpoint
+    { currentTip :: BlockHeader,
+        utxo :: UTxO,
+        discoveries :: (Discoveries s)
+    }
```
Comment (HeinrichApfelmus) [inline lib/core/src/Cardano/Wallet/DB/Checkpoints/Model.hs]: Please also use the "comma first" style here. 😅 (I suggest to not run the autoformatter on this module, just copy the file and rename the module with a dumb editor like `nano`.)

File: lib/core/src/Cardano/Wallet/DB/Checkpoints/Model.hs [hunk 7]
```diff
@@ -100,35 +120,73 @@ fromGenesis a = Checkpoints $ Map.singleton W.Origin a
 
 -- | Get the checkpoint with the largest 'SlotNo'.
 getLatest :: Checkpoints a -> (W.Slot, a)
-getLatest = from . Map.lookupMax . view #checkpoints 
-  where
-    from = fromMaybe (error "getLatest: there should always be at least a genesis checkpoint")
+getLatest = from . Map.lookupMax . view #checkpoints
+    where
+        from = fromMaybe (error "getLatest: there should always be at least a genesis checkpoint")
```
Comment (HeinrichApfelmus) [inline lib/core/src/Cardano/Wallet/DB/Checkpoints/Model.hs]:
```suggestion
  where
    from = fromMaybe (error "getLatest: there should always be at least a genesis checkpoint")
```

We use a style where `where` is special, it only gets 2 spaces of indentation. 🤓

File: lib/core/src/Cardano/Wallet/DB/Checkpoints/Model.hs [hunk 8]
```diff
@@ -100,35 +120,73 @@ fromGenesis a = Checkpoints $ Map.singleton W.Origin a
...
     | RollbackTo W.Slot
-        -- Rolls back to the latest checkpoint at or before this slot.
-    | RestrictTo [W.Slot]
-        -- ^ Restrict to the intersection of this list with
+    | -- Rolls back to the latest checkpoint at or before this slot.
+
+        -- | Restrict to the intersection of this list with
         -- the checkpoints that are already present.
         -- The genesis checkpoint will always be present.
+        RestrictTo [W.Slot]
```
Comment (HeinrichApfelmus) [inline lib/core/src/Cardano/Wallet/DB/Checkpoints/Model.hs]: Comments got mangled.

File: lib/core/src/Cardano/Wallet/DB/Wallets/Store.hs [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+import Data.Generics.Internal.VL
+    ( view, (^.) )
+import Database.Persist.Sql
+    ( Entity, SqlPersistT, deleteWhere, entityVal, selectList, (==.) )
+import Prelude
+
+import qualified Cardano.Wallet.Primitive.Types as W
+import qualified Data.Map.Strict as Map
+
+ {-------------------------------------------------------------------------------
```
Comment (HeinrichApfelmus) [inline lib/core/src/Cardano/Wallet/DB/Wallets/Store.hs]:
```suggestion
{-------------------------------------------------------------------------------
```

File: lib/core/src/Cardano/Wallet/DB/Wallets/Store.hs [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+-- | Store for 'WalletState' of a single wallet.
+mkStoreWallet
+    :: forall s. PersistAddressBook s
+    => W.WalletId
+    -> Store (SqlPersistT IO) (DeltaWalletState s)
+mkStoreWallet wid =
+    Store{ loadS = load, writeS = write, updateS = \_ -> update }
+  where
+    storeCheckpoints = mkStoreCheckpoints wid
+    -- storeTransactions = mkStoreTransactions wid
```
Comment (HeinrichApfelmus) [inline lib/core/src/Cardano/Wallet/DB/Wallets/Store.hs]:
```suggestion
    -- TODO: storeTransactions = mkStoreTransactions wid
```

Makes it more clear that it's not a piece of code that was commented out, but a piece that is to be written in.

File: lib/core/src/Cardano/Wallet/DB/Wallets/Store.hs [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+
+    load = do
+        eprologue <- maybe
+            (Left $ toException ErrBadFormatAddressPrologue) Right
+                <$> loadPrologue wid
+        echeckpoints <- loadS storeCheckpoints
+        -- etransactions <- loadS storeTransactions
```
Comment (HeinrichApfelmus) [inline lib/core/src/Cardano/Wallet/DB/Wallets/Store.hs]:
```suggestion
        -- TODO: etransactions <- loadS storeTransactions
```
as above

File: lib/core/src/Cardano/Wallet/DB/Wallets/Store.hs [hunk 12]
```diff
@@ -0,0 +1,128 @@
...
+         -- first update in list is last to be applied!
+        mapM_ update1 . reverse
+    update1 (ReplacePrologue prologue) =
+        insertPrologue wid prologue
+    update1 (UpdateCheckpoints delta) =
+        -- FIXME LATER during ADP-1043: remove 'undefined'
+        updateS storeCheckpoints undefined delta
+    -- update1 (UpdateTransactions delta) =
+    --     -- FIXME LATER during ADP-1043: remove 'undefined'
+    --     updateS storeTransactions undefined delta
```
Comment (HeinrichApfelmus) [inline lib/core/src/Cardano/Wallet/DB/Wallets/Store.hs]:
```suggestion
-- TODO
--  update1 (UpdateTransactions delta) =
--     -- FIXME LATER during ADP-1043: remove 'undefined'
--     updateS storeTransactions undefined delta
```

* Add TODO
* Identation no longer confusing with respect to existing code

**Review by HeinrichApfelmus — COMMENTED** [review]
A couple of style issues still. 😅

**Review by HeinrichApfelmus — APPROVED** [review]
Looks good now! 😊

I'd suggest to squash the commit history a little bit. For example, I removed the accidental changes to Transaction.hs (that autoformatter again. 😅 )

File: lib/core/src/Cardano/Wallet/DB/Checkpoints/Model.hs [hunk 13]
```diff
@@ -8,38 +9,54 @@
...
+    ) where
 
+import Cardano.Wallet.DB.Checkpoints.AddressBook
+    ( AddressBookIso (Discoveries, Prologue), addressIso )
+import qualified Cardano.Wallet.Primitive.Model as W
+import Cardano.Wallet.Primitive.Types
+    ( BlockHeader )
+import qualified Cardano.Wallet.Primitive.Types as W
+import Cardano.Wallet.Primitive.Types.UTxO
+    ( UTxO )
```
Comment (jonathanknowles) [inline lib/core/src/Cardano/Wallet/DB/Checkpoints/Model.hs]: Hi @paolino 

In our [coding standards document](https://github.com/input-output-hk/adrestia/blob/master/docs/code/Coding-Standards.md), it specifies to [separate imports into three distinct groups](https://github.com/input-output-hk/adrestia/blob/master/docs/code/Coding-Standards.md#stylish-haskell-is-used-to-format-grouped-imports--language-pragmas):

> In order to maximize readability, imports should be grouped into three groups, separated by a blank newline.
>
>   Prelude import
>   Explicit imports
>   Qualified imports

We've used this style throughout the wallet, so that when people look for a given import, they can search for imports in a predictable way.

Sorry to ask, but would you be able to go through this PR and make sure that each of the modules conforms to this rule?

Many thanks!

File: lib/core/src/Cardano/Wallet/DB/Checkpoints/Model.hs [hunk 14]
```diff
@@ -74,23 +91,26 @@ is clear that the data cannot exist at the genesis point
...
+        Checkpoints
 -------------------------------------------------------------------------------}
+
 -- | Collection of checkpoints indexed by 'Slot'.
 newtype Checkpoints a = Checkpoints
-    { checkpoints :: Map W.Slot a
-    -- ^ Map of checkpoints. Always contains the genesis checkpoint.
-    } deriving (Eq,Show,Generic)
+    { -- | Map of checkpoints. Always contains the genesis checkpoint.
+        checkpoints :: Map W.Slot a
```
Comment (jonathanknowles) [inline lib/core/src/Cardano/Wallet/DB/Checkpoints/Model.hs]: Hi @paolino 

Is there any reason to change the style of haddock comment here? It seemed fine before.

I think this kind of change should be avoided, _unless_ the code was breaking our coding standard in some way.

In the wallet we generally use the style where the haddock comment for a record field follows the field itself:

```suggestion
    { checkpoints :: Map W.Slot a
    -- ^ Map of checkpoints. Always contains the genesis checkpoint.
```
In some cases (typically for larger records with long comments), we indent each comment by one block (4 spaces):
```suggestion
    { checkpoints :: Map W.Slot a
        -- ^ Map of checkpoints. Always contains the genesis checkpoint.
```
Would it be possible to go through this PR and make sure that all changed/added record fields are of this form?

Many thanks!

File: lib/core/src/Cardano/Wallet/DB/Checkpoints/Model.hs [hunk 15]
```diff
@@ -100,35 +120,74 @@ fromGenesis a = Checkpoints $ Map.singleton W.Origin a
...
+    err = error
+        "getLatest: there should always be at least a genesis checkpoint"
 
 -- | Find the nearest 'Checkpoint' that is either at the given point or before.
 findNearestPoint :: Checkpoints a -> W.Slot -> Maybe W.Slot
 findNearestPoint m key = fst <$> Map.lookupLE key (view #checkpoints m)
 
 {-------------------------------------------------------------------------------
-    Delta type for Checkpoints
+        Delta type for Checkpoints
```
Comment (jonathanknowles) [inline lib/core/src/Cardano/Wallet/DB/Checkpoints/Model.hs]: Hi @paolino 

Is there a reason to make this change?

If not, would it be possible to remove this change, and others like it?

I think it's completely fair game to reformat things if they break our coding standards, but otherwise, I think we should refrain from making changes unless they directly relate to a PR.

File: lib/core/src/Cardano/Wallet/DB/Checkpoints/Model.hs [hunk 16]
```diff
@@ -100,35 +120,74 @@ fromGenesis a = Checkpoints $ Map.singleton W.Origin a
...
     type Base (DeltaCheckpoints a) = Checkpoints a
     apply (PutCheckpoint pt a) = over #checkpoints $ Map.insert pt a
-    apply (RollbackTo pt) = over #checkpoints $
-        Map.filterWithKey (\k _ -> k <= pt)
+    apply (RollbackTo pt) =
+        over #checkpoints $
+            Map.filterWithKey (\k _ -> k <= pt)
     apply (RestrictTo pts) = over #checkpoints $ \m ->
-        Map.restrictKeys m $ Set.fromList (W.Origin:pts)
+        Map.restrictKeys m $ Set.fromList (W.Origin : pts)
```
Comment (jonathanknowles) [inline lib/core/src/Cardano/Wallet/DB/Checkpoints/Model.hs]: Hi @paolino 

I also happen to prefer extra space around the `:` operator (I find it more readable), but AFAICT the existing style doesn't break our coding standards.

Is there any reason to make this change?

If it doesn't relate to the PR's purpose, would it be possible to remove changes like this from the PR?

File: lib/core/src/Cardano/Wallet/DB/Checkpoints/Model.hs [hunk 17]
```diff
@@ -100,35 +120,74 @@ fromGenesis a = Checkpoints $ Map.singleton W.Origin a
...
+-------------------------------------------------------------------------------}
+
+-- | Data stored in a single checkpoint.
+-- Only includes the 'UTxO' and the 'Discoveries', but not the 'Prologue'.
+data WalletCheckpoint s = WalletCheckpoint
+    {   currentTip :: BlockHeader
+    ,   utxo :: UTxO
+    ,   discoveries :: (Discoveries s)
+    }
+    deriving (Generic)
```
Comment (jonathanknowles) [inline lib/core/src/Cardano/Wallet/DB/Checkpoints/Model.hs]: In the wallet, we typically format records with just one space after `{` or `,`:

```suggestion
data WalletCheckpoint s = WalletCheckpoint
    { currentTip :: BlockHeader
    , utxo :: UTxO
    , discoveries :: (Discoveries s)
    }
    deriving (Generic)
```

Would it be possible to go through your PR and make sure that all changed or newly-added records conform to our existing style?

File: lib/core/src/Cardano/Wallet/DB/Layer.hs [hunk 18]
```diff
@@ -0,0 +1,813 @@
...
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE OverloadedLabels #-}
+{-# LANGUAGE RankNTypes #-}
+
+{-# LANGUAGE ScopedTypeVariables #-}
+
+{-# LANGUAGE TypeApplications #-}
```
Comment (jonathanknowles) [inline lib/core/src/Cardano/Wallet/DB/Layer.hs]:
```suggestion
{-# LANGUAGE RankNTypes #-}
{-# LANGUAGE ScopedTypeVariables #-}
{-# LANGUAGE TypeApplications #-}
```

File: lib/core/src/Cardano/Wallet/DB/Layer.hs [hunk 19]
```diff
@@ -0,0 +1,813 @@
...
+        {-----------------------------------------------------------------------
+                                       Keystore
+        -----------------------------------------------------------------------}
+
+        , putPrivateKey = \wid key -> ExceptT $
+            selectWallet wid >>= \case
+        Nothing -> pure $ Left $ ErrNoSuchWallet wid
+        Just _ -> Right <$> do
+            deleteWhere [PrivateKeyWalletId ==. wid]
+            insert_ (mkPrivateKeyEntity wid key)
```
Comment (jonathanknowles) [inline lib/core/src/Cardano/Wallet/DB/Layer.hs:777]: Hi @paolino

Is there any reason for the reformatting of this block? It makes the case statement quite hard to understand, because the case matches are indented **_to the left_** of the case. But typically in the wallet, we indent case matches one block (4 spaces) **_to the right_** of a case.

The original code was formatted like this:
```hs
          , putPrivateKey = \wid key -> ExceptT $ do
              selectWallet wid >>= \case
                  Nothing -> pure $ Left $ ErrNoSuchWallet wid
                  Just _ -> Right <$> do
                      deleteWhere [PrivateKeyWalletId ==. wid]
                      insert_ (mkPrivateKeyEntity wid key)
```

Notice that the `Nothing` is indented **_to the right_** of the case.

The new version is formatted like this:
```hs
          , putPrivateKey = \wid key -> ExceptT $ do
              selectWallet wid >>= \case
          Nothing -> pure $ Left $ ErrNoSuchWallet wid
          Just _ -> Right <$> do
              deleteWhere [PrivateKeyWalletId ==. wid]
              insert_ (mkPrivateKeyEntity wid key)
```
This seems to be a bit broken.

Please would it be possible to go through this PR and remove any reformattings of already-correctly-formatted code?

Many thanks!

File: lib/core/src/Cardano/Wallet/DB/Layer.hs [hunk 20]
```diff
@@ -0,0 +1,813 @@
...
+
+        {-----------------------------------------------------------------------
+                                 Delegation Rewards
+        -----------------------------------------------------------------------}
+
+        , putDelegationRewardBalance =
+            \wid amt -> ExceptT $
+            selectWallet wid >>= \case
+        Nothing -> pure $ Left $ ErrNoSuchWallet wid
+        Just _  -> Right <$> repsert
```
Comment (jonathanknowles) [inline lib/core/src/Cardano/Wallet/DB/Layer.hs:795]: Similar to my previous comment.

```suggestion
            selectWallet wid >>= \case
                Nothing -> pure $ Left $ ErrNoSuchWallet wid
                Just _  -> Right <$> repsert
```

File: lib/core/src/Cardano/Wallet/DB/Sqlite/WrapSTM.hs [hunk 21]
```diff
@@ -0,0 +1,83 @@
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+module Cardano.Wallet.DB.Sqlite.WrapSTM where
```
Comment (jonathanknowles) [inline lib/core/src/Cardano/Wallet/DB/Sqlite/WrapSTM.hs]:
```suggestion
{-# OPTIONS_GHC -Wno-orphans #-}

module Cardano.Wallet.DB.Sqlite.WrapSTM where
```

File: lib/core/src/Cardano/Wallet/DB/Transactions/Select.hs [hunk 22]
```diff
@@ -0,0 +1,360 @@
...
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedLabels #-}
+{-# LANGUAGE TupleSections #-}
+
+module Cardano.Wallet.DB.Transactions.Select where
+
+
+import Cardano.DB.Sqlite
+    ( chunkSize )
+import Cardano.Wallet.DB.Sqlite.Schema
```
Comment (jonathanknowles) [inline lib/core/src/Cardano/Wallet/DB/Transactions/Select.hs]: Hi @paolino 

Our [coding standards document](https://github.com/input-output-hk/adrestia/blob/master/docs/code/Coding-Standards.md) specifies to use [explicit imports by default](https://github.com/input-output-hk/adrestia/blob/master/docs/code/Coding-Standards.md#we-use-explicit-imports-by-default-and-favor-qualified-imports-for-ambiguous-functions).

```suggestion
import Cardano.Wallet.DB.Sqlite.Schema
    (add things here)
```

Could you go through this PR and make sure that we are not adding any other implicit imports?

Many thanks!

File: lib/core/src/Cardano/Wallet/DB/Transactions/Select.hs [hunk 23]
```diff
@@ -0,0 +1,360 @@
...
+selectTxMeta
+    :: W.WalletId
+    -> W.Hash "Tx"
+    -> SqlPersistT IO (Maybe TxMeta)
+selectTxMeta wid tid =
+    fmap entityVal <$> selectFirst
+        [ TxMetaWalletId ==. wid, TxMetaTxId ==. (TxId tid)]
+        [ Desc TxMetaSlot ]
+
+
```
Comment (jonathanknowles) [inline lib/core/src/Cardano/Wallet/DB/Transactions/Select.hs]:
```suggestion

```

**Review by jonathanknowles — CHANGES_REQUESTED** [review]
Hi @paolino 

Thanks for making this PR.

However, there are a few formatting issues introduced by this PR that I think we should address. Would it be possible to fix before merging? I've highlighted instances of these issues in the comments. (See our [coding standards](https://github.com/input-output-hk/adrestia/blob/master/docs/code/Coding-Standards.md) document for more info.)

In general, I would like to suggest that in future we split up PRs like this into two or more PRs. For example:

- PR # 1: Move modules to new locations, but don't make any code changes (unless it's really unavoidable).
- PR # 2: Split modules that need to be split, again without making any code changes (unless it's really unavoidable).
- PR # 3: Make actual change code changes to modules.

This way, reviewers can first review the location changes (which are easy), then review the module splittings (easy), and then in a later PR review actual changes to modules. This helps to avoid context switching between reviewing module renamings and reviewing changes to modules.

This PR both moves modules and makes changes to the content of those modules, which increases the burden of the reviewer, because it's now much harder to compare the old with the new.

What do you think?

Many thanks again for making this PR.

File: lib/core/src/Cardano/Wallet/DB/Checkpoints/Model.hs (hunk 13 above)
Comment (paolino) [reply lib/core/src/Cardano/Wallet/DB/Checkpoints/Model.hs]: Yes, it slipped through, sorry

File: lib/core/src/Cardano/Wallet/DB/Checkpoints/Model.hs (hunk 15 above)
Comment (paolino) [reply lib/core/src/Cardano/Wallet/DB/Checkpoints/Model.hs]: Reverting, it has no reason to be changed

File: lib/core/src/Cardano/Wallet/DB/Checkpoints/Model.hs (hunk 17 above)
Comment (paolino) [reply lib/core/src/Cardano/Wallet/DB/Checkpoints/Model.hs]: Will do

File: lib/core/src/Cardano/Wallet/DB/Checkpoints/Model.hs (hunk 16 above)
Comment (paolino) [reply lib/core/src/Cardano/Wallet/DB/Checkpoints/Model.hs]: Done!

File: lib/core/src/Cardano/Wallet/DB/Layer.hs (hunk 20 above)
Comment (paolino) [reply lib/core/src/Cardano/Wallet/DB/Layer.hs:795]: strangely enough the module is a trove of that *cases*. I tried to stay in style with it, even if I do not like it.
